packages/babel-parser/src/parser/util.js

```diff
@@ -111,6 +111,13 @@ export default class UtilParser extends Tokenizer {
     this.eat(type) || this.unexpected(pos, type);
   }
 
+  // Throws if the current token and the prev one are separated by a space.
+  assertNoSpace(message: string = "Unexpected space."): void {
+    if (this.state.start > this.state.lastTokEnd) {
+      this.raise(this.state.start, message);
```

Comment:
Should we have the error at the position where the space starts (this.state.lastTokEnd)?